distribution/std-lib/Standard/src/Image/Data/Matrix.enso

```diff
@@ -0,0 +1,150 @@
+
+    ## Calculates the per-element difference of two matrices or of a matrix and a scalar.
+
+       > Example
+         Subtract 10 from each channel of the matrix.
+             m - 10
+         Subtract 2 from second channel of the matrix.
+             m - [0, 2] == m - [0, 2, 0, 0]
+         Subtract two matrices:
+             m - m
+    - : (Number | Matrix | Vector) -> Matrix
+    - value = Internal.core_op this.opencv_mat value (Java_Matrix.subtract _ _ _)
+
+    ## Calculates the per-element product of two matrices or a matrix and a scalar.
+
+       > Example
+         Multiply each channel of the matrix by 2.
+             m * 2
+         Multiply first channel of the matrix by 2, and second channel by 4.
+             m * [2, 4, 1, 1]
+         Multiply first channel of the matrix by 2, and zero the rest of the channels.
+             m * [2]
+         Multiply two matrices.
+             m * m
+    * : (Number | Matrix | Vector) -> Matrix
+    * value = Internal.core_op this.opencv_mat value (Java_Matrix.multiply _ _ _)
+
+    ## Performs per-element division of two matrices or a matrix and a scalar.
+
+        > Example
+          Divide each channel of the matrix by 2.
+              m / 2
+          Divide first channel of the matrix by 2, and second channel by 4.
+              m / [2, 4, 1, 1]
+          Divide two matrices.
+              m / m
+    / : (Number | Matrix | Vector) -> Matrix
+    / value = Internal.core_op this.opencv_mat value (Java_Matrix.divide _ _ _)
+
+    ## Check the equality of two matrices.
+    == : Matrix -> Boolean
+    == that = Java_Matrix.is_equals this.opencv_mat that.opencv_mat
+
+    ## Convert this matrix to an image.
+    to_image : Image.Image
+    to_image = Image.Image (Image.from_vector this.to_vector this.rows this.channels)
+
+    ## Get the elemets of this matrix as a vector.
+    to_vector : Vector
+    to_vector =
+        arr = Java_Matrix.to_vector this.opencv_mat
+        Vector.Vector arr
+
+    ## Convert this matrix to Json.
+    to_json : Json
+    to_json = Json.String this.opencv_mat.to_text
+
+## Create the matrix with all elements - zeros.
+
+   > Example
+     Create a matrix.
+         Matrix.zeros 2 2 . to_vector == [0, 0, 0, 0]
+zeros : Integer -> Integer -> Integer -> Matrix
+zeros rows columns channels=1 =
+    Matrix (Java_Matrix.zeros rows columns channels)
+
+## Create the matrix with all elements - ones.
+
+   > Example
+     Create a matrix.
+         Matrix.ones 2 2 channels=2 . to_vector == [1, 1, 1, 1, 1, 1, 1, 1]
+ones : Integer -> Integer -> Integer -> Matrix
+ones rows columns channels=1 =
+    Matrix (Java_Matrix.ones rows columns channels)
+
+## Create an identity matrix containing ones on a main diagonal.
+
+   > Example
+     Create a matrix.
+         Matrix.eye 2 2 channels=1 . to_vector == [1, 0, 0, 1]
+eye : Integer -> Integer -> Integer -> Matrix
+eye rows columns channels=1 =
```

Comment:
I think it would be clearer to call this `identity`. I know that `eye` is used in some packages, but `identity` is far clearer overall.